tensorflow_datasets/core/utils/gpath.py

```diff
@@ -89,6 +89,16 @@ def resolve(self: _P, strict: bool = False) -> _P:
     """Returns the abolute path."""
     return self._new(posixpath.abspath(self._path_str))
 
+  def rglob(self: _P, pattern: str) -> Iterator[_P]:
+    """
+    Yielding all matching files (of any kind) recursively.
+    """
+    yield from self.glob(pattern)
+    if '**' in pattern:
+      for f in self.iterdir():
+        if f.is_dir():
+          yield from f.rglob(pattern)
+
   def glob(self: _P, pattern: str) -> Iterator[_P]:
```

Comment:
To be compatible with `pathlib.Path` API, `Path().glob('**/*.jpeg')` should also be recursive. See https://docs.python.org/3/library/pathlib.html#pathlib.Path.glob for doc.

Reply:
I'm not sure I understand what you mean to say over here. Currently, `Gpath().glob('**/*.jpeg)` does work as intended. At the beginning of the function, it uses `self.glob()` to find matching patterns in the given path (which are recursively called at the bottom).

Reply:
You could have a look at the `glob.iglob` implementation to see how they are doing it. There might be smarter way to do this

Reply:
Is there any reason to not directly use something like `pathlib.Path().rglob(pattern)` and pass it to a for loop to make it our desired type instead? The implementation of `glob.iglob` can get a bit complecated as we are trying to use `Gpath(path).rglob(pattern)` which has this exrta `path`, which is not present in `glob.glob()`.

Reply:
I'm not sure I understand. `pathlib.Path().rglob(pattern)` won't works with GCS right ?
`GPath('gs://some-path/').rglob('*.jpg')`

Reply:
Oh got it, I had no idea about that.

Reply:
I'll see if i can do a better implementation, until then I'll send a commit which is supposed to work properly in all the cases.